### ISSUE TITLE
Refactor and organize the Gemfile

### ISSUE THREAD

**Issue by hortongn** [uclibs/scholar_uc#253]
For some reason, doing a `bundle install` on scholar_uc causes the Gemfile.lock file to be changed.  Several gem versions are updated.  This is not normal behavior.  I've ruled out problems between bundler versions and running bundler on different platforms (Mac, Linux, etc.).  Other Hydra apps behave normally.  So there is something about the way the Gemfile is set up that is causing the issue.  A good start to fixing this is an audit of the gems being included and better organizing into groups for development/testing/production.

**Comment by hortongn** [comment]
I tested running "bundle install" on a branch of the scholar_uc app for over a month.  The Gemfile.lock never changed.  However, once I rebased my branch with develop and ran bundle install again, the Gemfile.lock file had many changes.  The only change was that the curate gem version specified in the Gemfile was updated to a new Git ref.  So it's looking like whenever we point the Gemfile to a new version of our curate_fork, bundler will update the gem's dependencies as well.  Even if we peg all the dependencies to specific versions, we have no control over the dependencies of each dependency.
